src/node.cc

```diff
@@ -1197,6 +1224,18 @@ enum encoding ParseEncoding(Isolate* isolate,
   }
 }
 
+
+enum encoding ParseEncoding(Isolate* isolate,
+                            Handle<Value> encoding_v,
+                            enum encoding _default) {
+  if (!encoding_v->IsString())
+    return _default;
+
+  node::Utf8Value encoding(isolate, encoding_v);
+
+  return ParseEncoding(*encoding, _default);
```

Comment:
I'm +1 for JS-land thing, but for now - let's do it the way it is. `encoding->length()` doesn't seem to be improving situation much on my benchmarks.